editor/effects.js

```diff
@@ -273,4 +284,106 @@ export default {
+				dispatch( {
+					type: 'FETCH_REUSABLE_BLOCKS_SUCCESS',
+					reusableBlocks: reusableBlocks.map( ( { id, name, content } ) => {
+						const [ { name: type, attributes } ] = parse( content );
+						return { id, name, type, attributes };
+					} ),
+				} );
+			},
+			( error ) => {
+				dispatch( {
+					type: 'FETCH_REUSABLE_BLOCKS_FAILURE',
+					error: get( error, 'responseJSON', {
+						code: 'unknown_error',
+						message: __( 'An unknown error occurred.' ),
+					} ),
+				} );
+			}
+		);
+	},
+	FETCH_REUSABLE_BLOCKS_SUCCESS( action, store ) {
+		const { reusableBlocks } = action;
+		const { dispatch } = store;
+
+		dispatch( addReusableBlocks( reusableBlocks ) );
+	},
+	FETCH_REUSABLE_BLOCK( action, store ) {
+		const { id } = action;
+		const { dispatch } = store;
+
+		new wp.api.models.ReusableBlocks( { id } ).fetch().then(
+			( { name, content } ) => {
+				const [ { name: type, attributes } ] = parse( content );
+				dispatch( {
+					type: 'FETCH_REUSABLE_BLOCK_SUCCESS',
+					reusableBlock: { id, name, type, attributes },
+				} );
+			},
+			( error ) => {
+				dispatch( {
+					type: 'FETCH_REUSABLE_BLOCK_FAILURE',
+					error: get( error, 'responseJSON', {
+						code: 'unknown_error',
+						message: __( 'An unknown error occurred.' ),
+					} ),
+				} );
+			}
+		);
+	},
+	FETCH_REUSABLE_BLOCK_SUCCESS( action, store ) {
+		const { reusableBlock } = action;
+		const { dispatch } = store;
+
+		dispatch( addReusableBlocks( reusableBlock ) );
+	},
+	SAVE_REUSABLE_BLOCK( action, store ) {
+		const { id } = action;
+		const { getState, dispatch } = store;
+
+		const { name, type, attributes } = getReusableBlock( getState(), id );
+		const content = serialize( createBlock( type, attributes ) );
+
+		new wp.api.models.ReusableBlocks( { id, name, content } ).save().then(
+			() => {
+				dispatch( {
+					type: 'SAVE_REUSABLE_BLOCK_SUCCESS',
+				} );
+			},
+			( error ) => {
+				dispatch( {
+					type: 'SAVE_REUSABLE_BLOCK_FAILURE',
+					error: get( error, 'responseJSON', {
+						code: 'unknown_error',
+						message: __( 'An unknown error occurred.' ),
+					} ),
+				} );
+			}
+		);
+	},
+	ATTACH_BLOCK( action, store ) {
+		const { getState, dispatch } = store;
+
+		const oldBlock = getBlock( getState(), action.uid );
+		const reusableBlock = getReusableBlock( getState(), oldBlock.attributes.ref );
+		const newBlock = createBlock( reusableBlock.type, reusableBlock.attributes );
+		dispatch( replaceBlocks( [ oldBlock.uid ], [ newBlock ] ) );
+	},
+	DETACH_BLOCK( action, store ) {
+		const { getState, dispatch } = store;
+
+		const oldBlock = getBlock( getState(), action.uid );
+		const reusableBlock = createReusableBlock( oldBlock.name, oldBlock.attributes );
+		const newBlock = createBlock( 'core/reusable-block', { ref: reusableBlock.id } );
+		dispatch( addReusableBlocks( reusableBlock ) );
+		dispatch( saveReusableBlock( reusableBlock.id ) );
+		dispatch( replaceBlocks( [ oldBlock.uid ], [ newBlock ] ) );
+	},
```

Comment:
I'm confused, are these two actions reversed (attache/detach). Attach should create a reusable block and detach should create a regular block (as I understand it at least)

Reply:
Yeah, I wrote this all thinking that attach would create a regular block since you're taking a block that is _unattached_ from the post (i.e., it's saved separately from the post) and _attaching_ it to a post.

In my view, the whole attach terminology is confusing. I was thinking of making the actions `MAKE_BLOCK_REUSABLE` and `MAKE_BLOCK_STATIC`. Thoughts?

Reply:
> I was thinking of avoiding the issue altogether by making the actions MAKE_BLOCK_REUSABLE and MAKE_BLOCK_STATIC. Thoughts?

sounds good to me 👍

Reply:
Fixed in f1368ba4 📈